src/backend/executor/nodeModifyTable.c

```diff
@@ -1053,7 +1053,7 @@ ExecInsert(ModifyTableState *mtstate,
 	if (resultRelationDesc->rd_rel->relispartition)
 	{
 		mtstate->mt_leaf_relids_inserted =
-			bms_add_member(mtstate->mt_leaf_relids_inserted, RelationGetRelid(resultRelationDesc));
+			list_append_unique_oid(mtstate->mt_leaf_relids_inserted, RelationGetRelid(resultRelationDesc));
```

Comment:
If we insert N tuples into the partition table that has M children, the time may be O(N*M/nsegs). How will be the performance affected?